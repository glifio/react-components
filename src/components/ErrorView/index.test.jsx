import { render, act, fireEvent } from '@testing-library/react'
import { ErrorView } from '.'
import { Base } from './index.stories'

describe('Error', () => {
  test('renders the story', () => {
    const { container } = render(<Base {...Base.args} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders the error', () => {
    const { container } = render(
      <ErrorView
        title='Ledger only supports Chrome'
        description='Please install Google Chrome to continue using your Ledger device, or choose an alternative setup option'
        linkDisplay='Install Google Chrome.'
        linkhref='https://www.google.com/chrome'
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders the error with the correct text', () => {
    const { getByText } = render(
      <ErrorView
        title='Ledger only supports Chrome'
        description='Please install Google Chrome to continue using your Ledger device, or choose an alternative setup option'
        linkDisplay='Install Google Chrome.'
        linkhref='https://www.google.com/chrome'
      />
    )

    expect(getByText('Ledger only supports Chrome')).toBeTruthy()
    expect(
      getByText(
        'Please install Google Chrome to continue using your Ledger device, or choose an alternative setup option'
      )
    ).toBeTruthy()
    expect(getByText('Install Google Chrome.')).toBeTruthy()
  })

  test.skip('clicking "Back" takes you back to onboarding', () => {
    const useRouter = jest.spyOn(
      /* eslint-disable-next-line global-require */
      require('next/router'),
      'useRouter'
    )
    const mockRouterReplace = jest.fn(() => {})
    useRouter.mockImplementationOnce(() => ({
      replace: mockRouterReplace,
      query: 'network=t'
    }))

    const { getByText } = render(
      <ErrorView
        title='Ledger only supports Chrome'
        description='Please install Google Chrome to continue using your Ledger device, or choose an alternative setup option'
        linkDisplay='Install Google Chrome.'
        linkhref='https://www.google.com/chrome'
      />
    )

    act(() => {
      fireEvent.click(getByText('Back'))
    })

    expect(mockRouterReplace).toHaveBeenCalled()
    expect(mockRouterReplace).toHaveBeenCalledWith('/')
  })
})
