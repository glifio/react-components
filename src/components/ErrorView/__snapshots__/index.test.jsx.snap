// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders the error 1`] = `
.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  white-space: nowrap;
  cursor: pointer;
  gap: 0.75rem;
  padding: 0.5em 0.75em;
  color: var(--black);
  border-width: 1px;
  border-style: solid;
  border-radius: 8px;
  border-color: var(--black);
  background: transparent;
}

.c5 svg path {
  stroke: var(--black);
}

.c5:hover {
  color: var(--purple-medium);
  border-color: var(--purple-medium);
}

.c5:hover svg path {
  stroke: var(--purple-medium);
}

.c5:active {
  color: var(--white) !important;
  border-color: var(--purple-medium) !important;
  background: var(--purple-medium) !important;
}

.c5:active svg path {
  stroke: var(--white) !important;
}

.c3 {
  vertical-align: middle;
}

.c1 {
  padding: 1.5em;
  border-radius: 8px;
  overflow: hidden;
  text-align: center;
  word-break: break-word;
}

.c1 > *:first-child {
  margin-top: 0;
}

.c1 > *:last-child {
  margin-bottom: 0;
}

.c1 > svg:first-child {
  display: block;
  margin: 0 auto 1em;
}

.c1 > h2:first-child {
  margin: 0;
}

.c1 header:first-child {
  margin: -1.5em -1.5em 1.5em -1.5em;
  padding: 1.5em;
  background-color: var(--purple-medium);
  color: var(--white);
}

.c2 {
  background-color: var(--red-light);
  color: var(--red-dark);
}

.c2 hr {
  border-color: var(--red-dark);
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: var(--space-m);
  margin: 1.5em 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: var(--space-m);
  width: 100%;
  margin: 0 auto;
  max-width: 40em;
}

.c0 form {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: var(--space-m);
}

<div
  class="c0"
>
  <div
    class="c1 c2"
  >
    <svg
      class="c3 has-stroke"
      fill="none"
      height="20"
      stroke="var(--red-dark)"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 22 20"
      width="22"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11 7.74982V9.99986M11 14.4999H11.0112M3.24282 19H18.7572C20.4815 19 21.5586 17.1246 20.6965 15.6249L12.9393 2.12474C12.0771 0.625088 9.92286 0.625088 9.0607 2.12474L1.30352 15.6249C0.441363 17.1246 1.5185 19 3.24282 19Z"
      />
    </svg>
    <h2>
      Ledger only supports Chrome
    </h2>
    <hr />
    <p>
      Please install Google Chrome to continue using your Ledger device, or choose an alternative setup option
    </p>
    <p>
      <a
        href="https://www.google.com/chrome"
        rel="noreferrer noopener"
        target="_blank"
      >
        Install Google Chrome.
      </a>
    </p>
    <hr />
    <p>
      Get help in 
      <a
        href="https://discord.gg/B9ju5Eu4Rq"
        rel="noreferrer noopener"
        target="_blank"
      >
        Discord
      </a>
       or hit us up on 
      <a
        href="https://twitter.com/glifio"
        rel="noreferrer noopener"
        target="_blank"
      >
        Twitter
      </a>
    </p>
  </div>
  <div
    class="c4"
  >
    <button
      class="c5"
    >
      Back
    </button>
  </div>
</div>
`;

exports[`Error renders the story 1`] = `
.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  white-space: nowrap;
  cursor: pointer;
  gap: 0.75rem;
  padding: 0.5em 0.75em;
  color: var(--black);
  border-width: 1px;
  border-style: solid;
  border-radius: 8px;
  border-color: var(--black);
  background: transparent;
}

.c5 svg path {
  stroke: var(--black);
}

.c5:hover {
  color: var(--purple-medium);
  border-color: var(--purple-medium);
}

.c5:hover svg path {
  stroke: var(--purple-medium);
}

.c5:active {
  color: var(--white) !important;
  border-color: var(--purple-medium) !important;
  background: var(--purple-medium) !important;
}

.c5:active svg path {
  stroke: var(--white) !important;
}

.c3 {
  vertical-align: middle;
}

.c1 {
  padding: 1.5em;
  border-radius: 8px;
  overflow: hidden;
  text-align: center;
  word-break: break-word;
}

.c1 > *:first-child {
  margin-top: 0;
}

.c1 > *:last-child {
  margin-bottom: 0;
}

.c1 > svg:first-child {
  display: block;
  margin: 0 auto 1em;
}

.c1 > h2:first-child {
  margin: 0;
}

.c1 header:first-child {
  margin: -1.5em -1.5em 1.5em -1.5em;
  padding: 1.5em;
  background-color: var(--purple-medium);
  color: var(--white);
}

.c2 {
  background-color: var(--red-light);
  color: var(--red-dark);
}

.c2 hr {
  border-color: var(--red-dark);
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: var(--space-m);
  margin: 1.5em 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: var(--space-m);
  width: 100%;
  margin: 0 auto;
  max-width: 40em;
}

.c0 form {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: var(--space-m);
}

<div
  class="c0"
>
  <div
    class="c1 c2"
  >
    <svg
      class="c3 has-stroke"
      fill="none"
      height="20"
      stroke="var(--red-dark)"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 22 20"
      width="22"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11 7.74982V9.99986M11 14.4999H11.0112M3.24282 19H18.7572C20.4815 19 21.5586 17.1246 20.6965 15.6249L12.9393 2.12474C12.0771 0.625088 9.92286 0.625088 9.0607 2.12474L1.30352 15.6249C0.441363 17.1246 1.5185 19 3.24282 19Z"
      />
    </svg>
    <h2>
      Ledger only supports Chrome
    </h2>
    <hr />
    <p>
      Please install Google Chrome to continue using your Ledger device, or choose an alternative setup option
    </p>
    <p>
      <a
        href="https://www.google.com/chrome"
        rel="noreferrer noopener"
        target="_blank"
      >
        Install Google Chrome.
      </a>
    </p>
    <hr />
    <p>
      Get help in 
      <a
        href="https://discord.gg/B9ju5Eu4Rq"
        rel="noreferrer noopener"
        target="_blank"
      >
        Discord
      </a>
       or hit us up on 
      <a
        href="https://twitter.com/glifio"
        rel="noreferrer noopener"
        target="_blank"
      >
        Twitter
      </a>
    </p>
  </div>
  <div
    class="c4"
  >
    <button
      class="c5"
    >
      Back
    </button>
  </div>
</div>
`;
